core/src/worker/workflow/wft_poller.rs

```diff
@@ -121,26 +111,63 @@ impl WFTPollerShared {
             {
                 let mut sticky_active = sticky_active.clone();
                 let mut non_sticky_active = non_sticky_active.clone();
+                let mut sticky_backlog = self.last_seen_sticky_backlog.0.clone();
+
                 loop {
                     let num_sticky_active = *sticky_active.borrow_and_update();
                     let num_non_sticky_active = *non_sticky_active.borrow_and_update();
-                    let both_are_zero = num_sticky_active == 0 && num_non_sticky_active == 0;
-                    if both_are_zero {
+                    let num_sticky_backlog = *sticky_backlog.borrow_and_update();
+
+                    let allow = || {
                         if !is_sticky {
-                            break;
+                            // There should always be at least one non-sticky poller.
+                            if num_non_sticky_active == 0 {
+                                return true;
+                            }
+
+                            // Do not allow an additional non-sticky poller to prevent starting a first sticky poller.
+                            if num_sticky_active == 0 && num_non_sticky_active + 1 >= max_slots {
+                                return false;
+                            }
+
+                            // If there's a meaningful sticky backlog, prioritize sticky.
+                            if num_sticky_backlog > 1 && num_sticky_backlog > num_sticky_active {
+                                return false;
+                            }
+                        } else {
+                            // There should always be at least one sticky poller.
+                            if num_sticky_active == 0 {
+                                return true;
+                            }
+
+                            // Do not allow an additional sticky poller to prevent starting a first non-sticky poller.
+                            if num_non_sticky_active == 0 && num_sticky_active + 1 >= max_slots {
+                                return false;
+                            }
+
+                            // If there's a meaningful sticky backlog, prioritize sticky.
+                            if num_sticky_backlog > 1 && num_sticky_backlog > num_sticky_active {
+                                return true;
+                            }
                         }
-                    } else {
-                        let would_exceed_max_slots =
-                            (num_sticky_active + num_non_sticky_active + 1) >= max_slots;
-                        let must_wait = would_exceed_max_slots
-                            && (num_sticky_active == 0 || num_non_sticky_active == 0);
-                        if !must_wait {
-                            break;
+
+                        // Just balance the two poller types.
+                        // FIXME: Do we need anything more here, to ensure proper balancing?
```

Comment:
Go effectively just forces there to be an even amount of both kinds of pending requests, except always favoring sticky during a backlog, and when counts are even: https://github.com/temporalio/sdk-go/blob/9682a692145f503a02b73f32435cb04bdc6f57c1/internal/internal_task_pollers.go#L782

We already are favoring it with a backlog. It's not obvious to me that just making an even amount is the best move either, but, it is what Core used to do and clearly it at least works, so we could go with that.